src/controls/qml/LayerStack.qml

```diff
@@ -56,11 +57,27 @@ Item {
         params["x"] = 0
         params["y"] = 0
         if(typeof firstPage != 'undefined' && firstPage.status === Component.Ready) {
-            var itm=firstPage.createObject(content, params)
+            itm=firstPage.createObject(content, params)
             itm.clip = true
         }
     }
 
+    onFirstPageChanged: {
+        if(typeof itm != 'undefined') {
+            itm.destroy()
+            var params = {}
+            params["width"] = Qt.binding(function() { return width })
+            params["height"] =  Qt.binding(function() { return height })
+            params["x"] = 0
+            params["y"] = 0
+            itm=firstPage.createObject(content, params)
+            itm.clip = true
```

Comment:
Also that is now quite a bit of duplicated code, maybe you can extract this into a function that gets called in both places ?

Reply:
I'm not entirely sure this is useful - it saves 3-5 lines of code and makes things a bit harder to understand. I can do it if you think it is appropriate, but I've just pushed a version that solves everything apart from this issue.